### PR TITLE
When SwiftPM supports `--experimental-prepare-for-indexing` pass it to the prepare command

### PR DESCRIPTION
**Pull request by ahoppen**
The main purpose for now is that this makes it easier for me to live on background indexing combined with https://github.com/apple/swift-package-manager/pull/7574.

### CODE REVIEW

**Comment by ahoppen** [comment]
@swift-ci Please test

File: Sources/SKSwiftPMWorkspace/SwiftPMBuildSystem.swift [hunk 1]
```diff
@@ -106,6 +106,45 @@ public actor SwiftPMBuildSystem {
...
+          }
+
+          do {
+            let process = Process(args: swift.pathString, "build", "--help")
+            try process.launch()
+            let result = try await process.waitUntilExit()
+            guard let output = String(bytes: try result.output.get(), encoding: .utf8) else {
+              return false
+            }
+            return output.contains("--experimental-prepare-for-indexing")
```
Comment (bnbarham) [inline Sources/SKSwiftPMWorkspace/SwiftPMBuildSystem.swift:135]: I actually didn't notice that it wasn't hidden from help. We should change that, in which case you will want `--help-hidden` above.

**Review by hamishknight — APPROVED** [review]

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Review by bnbarham — APPROVED** [review]

**Pull request merged**